chai-immutable.js

```diff
@@ -540,5 +540,97 @@
     assert.sizeOf = function (collection, expected) {
       new Assertion(collection).size(expected);
     };
+
+    /**
+     * ### .property(path, value)
+     *
+     * Asserts that property specified exists, and if a value specified, is equal
+     * to the given value.
+     *
+     * ```js
+     * var obj = Immutable.fromJS({
+     *   left: 1,
+     *   right: {
+     *     left: 2
+     *     right: 3
+     *   }
+     * });
+     *
+     * assert.property(obj, 'left');
+     * assert.propertyVal(obj, 'left', 1);
+     *
+     * assert.deepProperty(obj, ['right', 'right']);
+     * assert.deepPropertyVal(obj, ['right', 'right'], 3);
+     * ```
+     *
+     * @name property
+     * @param {path} either the single propery name or array used for `getIn()`
+     * @param {value} expected value found at that location
+     * @api public
+     */
+    function assertProperty(_super) {
+      return function (path, val, msg) {
+        var obj = this._obj;
+        var isDeep = Boolean(utils.flag(this, 'deep'));
+        var negate = Boolean(utils.flag(this, 'negate'));
+
+        var descriptor;
+        var hasProperty;
+        var value;
+
+        if (!Immutable.Iterable.isIterable(obj)) {
+          return _super.apply(this, arguments);
```

Comment:
Could you follow the same pattern as other assertions, that is, having this check as high in the function as possible and returning `_super.apply(this, arguments);` in an `else` statement at the very end?
Probably not the nicest way to do so, but at least consistent with the rest of the file.